src/GraphQL/GraphQLExtensions.cs

```diff
@@ -566,15 +566,25 @@ public static bool IsValidDefault(this IGraphType type, object value)
         /// </summary>
         public static IValue ToAST(this IGraphType type, object value)
         {
+            if (type == null)
+                throw new ArgumentNullException(nameof(type));
+
             if (type is NonNullGraphType nonnull)
             {
-                if (value == null)
+                var astValue = ToAST(nonnull.ResolvedType, value);
+
+                if (astValue is NullValue)
                     throw new InvalidOperationException($"Unable to get an AST representation of type '{nonnull}' for null value.");
```

Comment:
```suggestion
                    throw new InvalidOperationException($"Unable to get an AST representation of null value for type '{nonnull}'.");
```

I'm not sure how to say correctly. It seems to me that AST relates more to value than type.

Reply:
Maybe something like this?

```suggestion
                {
                    if (value == null)
                    {
                        throw new InvalidOperationException($"Null value provided for nonnull type '{nonnull}'.");
                    }
                    else
                    {
                        throw new InvalidOperationException($"Unable to get a nonnull AST representation of nonnull type '{nonnull}' for value '{value}'.");
                    }
                }
```

Reply:
I don't like messages like `nonnull type XXX!`. Because of them it always seems to me that there may be a `XXX!!` type.

Reply:
I updated my initial suggestion.